test/unit/mem/copy/src/BufSlicing.cpp

```diff
@@ -0,0 +1,543 @@
+    {
+        constexpr Idx extentsDimOne = 13;
+        constexpr Idx extentsDimTwo = 7;
+        constexpr Idx extentsDimThree = 5;
+
+        constexpr Idx extentsSubViewDimOne = 5;
+        constexpr Idx extentsSubViewDimTwo = 3;
+        constexpr Idx extentsSubViewDimThree = 3;
+
+        constexpr Idx offsetDimOne = 3;
+        constexpr Idx offsetDimTwo = 2;
+        constexpr Idx offsetDimThree = 1;
+
+        using Vec = alpaka::vec::Vec< Dim, Idx >;
+
+        using TestContainer =TestContainer<Dim, Idx, Acc, Data>;
+
+        using BufHost = typename TestContainer::BufHost;
+        using BufDevice = typename TestContainer::BufDevice;
+
+        TestContainer slicingTest;
+
+        // Setup extents, extentsSubView and offsets.
+        Vec extents =
+            DimHelper<
+                Dim,
+                Idx
+            >::getVec(
+                extentsDimOne,
+                extentsDimTwo,
+                extentsDimThree
+            );
+        Vec extentsSubView = DimHelper< Dim, Idx >::getVec(
+            extentsSubViewDimOne,
+            extentsSubViewDimTwo,
+            extentsSubViewDimThree
+        );
+        Vec offsets = DimHelper< Dim, Idx >::getVec(
+            offsetDimOne,
+            offsetDimTwo,
+            offsetDimThree
+        );
+        // This is the initial buffer.
+        BufHost indexedBuffer = slicingTest.createHostBuffer(
+            extents,
+            true
+        );
+        // This buffer will hold the sliced-buffer when it was copied to the host.
+        BufHost resultBuffer = slicingTest.createHostBuffer(
+            extentsSubView,
+            false
+        );
+
+        // Copy of the indexBuffer on the deviceSide.
+        BufDevice deviceBuffer = slicingTest.createDeviceBuffer( extents );
+
+        // Start: Main-Test
+        slicingTest.copyToAcc(
+            indexedBuffer,
+            deviceBuffer,
+            extents
+        );
+
+        BufDevice slicedBuffer = slicingTest.sliceOnDevice(
+            deviceBuffer,
+            extentsSubView,
+            offsets
+        );
+
+        slicingTest.copyToHost(
+            slicedBuffer,
+            resultBuffer,
+            extentsSubView
+        );
+
+        // resultBuffer will be compared with the manually computed results.
+        slicingTest.compareBuffer(
+            resultBuffer,
+            DimHelper< Dim, Idx >::template cmpCorrectlySlicedBuffer< Data >(
+                extents,
+                extentsSubView,
+                offsets
+            ),
+            extentsSubView
+        );
+    }
+};
+
+TEST_CASE( "memBufSlicingTest",
+    "[memBuf]" )
+{
+    /*
+     * FIXME
+     * Current call-chain works this way but does'nt look like it.
+     * Desired (or similar):
+     * ForEachDataType::call(
+     *     ForEachDimension::call(
+     *         ForEachAccelerator::call(
+     *             TestSlicing() ) ) )
+     */
+    ForEachDataType< size_t >( );
```

Comment:
By only testing `size_t` and not all the other possible index types you lose a lot of test coverage in comparison to the original test.